runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -127,6 +129,30 @@ decl_module! {
 	}
 }
 
+// At the point that this is run, the provisioner has already chosen a set of extrinsics,
+// so we can rely on calculations like block weight.
+fn inclusion_inherents_would_overload_block<T: 'static + frame_system::Config>() -> bool {
+	frame_system::Module::<T>::block_weight().total() > <T as frame_system::Config>::MaximumBlockWeight::get() - BLOCK_WEIGHT_MARGIN
```

Comment:
Instead of calculating a fixed margin, we could pretty easily redo this calculation as a fraction of total block weight, if desired.